pkg/csi/service/wcpguest/controller.go

```diff
@@ -1345,11 +1355,27 @@ func (c *controller) ListVolumes(ctx context.Context, req *csi.ListVolumesReques
 
 func (c *controller) GetCapacity(ctx context.Context, req *csi.GetCapacityRequest) (
 	*csi.GetCapacityResponse, error) {
-
 	ctx = logger.NewContextWithLogger(ctx)
 	log := logger.GetLogger(ctx)
 	log.Infof("GetCapacity: called with args %+v", *req)
-	return nil, status.Error(codes.Unimplemented, "")
+
+	var totalcapacity, maxvolumesize int64
+	zonesMap := commonco.ContainerOrchestratorUtility.GetZonesForNamespace(c.supervisorNamespace)
+	if req.AccessibleTopology != nil {
+		segments := req.AccessibleTopology.GetSegments()
+		if _, exists := zonesMap[segments["topology.csi.vmware.com/k8s-zone"]]; !exists {
```

Comment:
can we add brief comment on this code is handling zone removal case ?

Reply:
we can return `return nil, status.Error(codes.Unimplemented, "")` when FSS is disabled.

Reply:
added comment

```
// if zone is marked for removal or removed from namespace then zone will be removed from namespaceToZoneMap map
        namespaceToZoneMap = make(map[string]map[string]struct{})

```